Rollbar/RollbarLogger.cs

```diff
@@ -209,25 +219,55 @@ public ILogger Debug(object obj, IDictionary<string, object> custom = null)
 
         #endregion ILogger
 
-        private void Report(System.Exception e, ErrorLevel? level = ErrorLevel.Error, IDictionary<string, object> custom = null)
+        internal void Report(
+            System.Exception e, 
+            ErrorLevel? level = ErrorLevel.Error, 
+            IDictionary<string, object> custom = null, 
+            TimeSpan? timeout = null,
+            SemaphoreSlim signal = null
+            )
         {
             SendBodyAsync(new Body(e), level, custom);
         }
 
-        private void Report(string message, ErrorLevel? level = ErrorLevel.Error, IDictionary<string, object> custom = null)
+        internal void Report(
+            string message, 
+            ErrorLevel? level = ErrorLevel.Error, 
+            IDictionary<string, object> custom = null,
+            TimeSpan? timeout = null,
+            SemaphoreSlim signal = null
+            )
         {
-            SendBodyAsync(new Body(new Message(message)), level, custom);
+            SendBodyAsync(new Body(new Message(message)), level, custom, timeout, signal);
         }
 
-        private void SendBodyAsync(Body body, ErrorLevel? level, IDictionary<string, object> custom)
+        private void SendBodyAsync(
+            Body body, ErrorLevel? level, 
+            IDictionary<string, object> custom,
+            TimeSpan? timeout = null,
+            SemaphoreSlim signal = null
+            )
         {
+            DateTime? timeoutAt = null;
+            if (timeout.HasValue)
+            {
+                timeoutAt = DateTime.Now.Add(timeout.Value);
```

Comment:
What does the timeout mean if you set the `timeoutAt` time here? It seems like we use the `timeout` value a couple different ways.

Reply:
"timeout" is a time span specifying for how long to wait for sending out a payload.
"timeoutAt" is a cut off time to give up sending the payload and to discard the payload. it is calculated at the time of allocating the payload plus the "timeout". based on this timestamp the queues processing thread knows when to give up transmission of any particular payload.